boards/arm/wio_terminal/wio_terminal_defconfig

```diff
@@ -0,0 +1,16 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_SAMD51=y
+CONFIG_SOC_PART_NUMBER_SAMD51P19A=y
+CONFIG_SOC_ATMEL_SAMD5X_XOSC32K=y
+CONFIG_SOC_ATMEL_SAMD5X_XOSC32K_AS_MAIN=y
+CONFIG_BOARD_WIO_TERMINAL=y
+CONFIG_ARM_MPU=y
+CONFIG_CORTEX_M_SYSTICK=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_BOOTLOADER_BOSSA=y
+CONFIG_BOOTLOADER_BOSSA_ADAFRUIT_UF2=y
+CONFIG_USE_DT_CODE_PARTITION=y
+CONFIG_GPIO=y
+CONFIG_PINMUX=y
+CONFIG_REGULATOR=y
```

Comment:
What is this option? I see few board configure this, not sure it'a default choice to be added here?
I don't find clear documentation of the impact. Not described in the other boards for example...

Reply:
My opinion is also that most people dont' use Windows, so it's not useful. If you are using Windows you can add this config in your own project.

Reply:
> What is this option? I see few board configure this, not sure it'a default choice to be added here? I don't find clear documentation of the impact. Not described in the other boards for example...

This allows to get a .uf2 file in the build output directory, which is a convenient way to have a binary that can actually be installed by anyone, by means of putting the Wio Terminal in "boot mode" and copying the file on the mass storage. It is also useful on Windows as the bossa flashing tool is not available (but will soon be, though, see #37538).

Reply:
> My opinion is also that most people dont' use Windows, so it's not useful. If you are using Windows you can add this config in your own project.

Except Windows at least 50% of the user base, if not more :)
Even if not enabled by default, it might make a lot of sense to document the ability to turn on this flag in the README, as most people otherwise ignore that they can easily distribute a "flashable" app through this .uf2 format, and I think it is kind of nice for a platform such as Wio Terminal where people might end up sharing the app with people that are not tech-savvy (ex. artists, etc.)

Reply:
Okay I understand now the usage of the UF2. It's not Windows specific at all! The device mounts as a new disk on any OS when entering bootloader, and it's possible to copy/paste the UF2 file to it to flash the MCU.
And so I agree it's interesting to have it for this kind of boards :-)

However, I'm trying to generate the `zephyr.uf2` file with `CONFIG_BUILD_OUTPUT_UF2=y`, and it's not generated... it depends of `CONFIG_BUILD_OUTPUT_BIN` which is set, so it's not clear for me... Do you know if there is extra requirements ? The `.config` file  still have `# CONFIG_BUILD_OUTPUT_UF2 is not set` even after rebuild all the firmware...

Reply:
@kartben done